workers/unity/Packages/io.improbable.gdk.core/Worker/WorldCommandsReceivedStorage.cs

```diff
@@ -132,16 +124,12 @@ public void AddResponse(WorldCommands.EntityQuery.ReceivedResponse response)
             }
 
             var responseIndex = reserveEntityIdsResponses.GetResponseIndex(requestId);
-            if (responseIndex < 0)
-            {
-                return ReceivedMessagesSpan<WorldCommands.ReserveEntityIds.ReceivedResponse>.Empty();
-            }
-
-            return new ReceivedMessagesSpan<WorldCommands.ReserveEntityIds.ReceivedResponse>(reserveEntityIdsResponses,
-                responseIndex, 1);
+            return !responseIndex.HasValue
```

Comment:
nit: would be good to also invert it here and below

Reply:
Agreed, forgot about the hand made ones